### PR TITLE
feat(planner/static): Support hugo configuration directory mode & test case

### PR DESCRIPTION
**Pull request by MashiroWang**
image

Enhance the detection of the Hugo framework by searching in `config/_default`

### CODE REVIEW

File: internal/static/identify.go [hunk 1]
```diff
@@ -32,6 +39,16 @@ func (i *identify) PlanMeta(options plan.NewPlannerOptions) types.PlanMeta {
 		return types.PlanMeta{"framework": "hugo"}
 	}
 
+	// Trying to locate the target file in the configuration directory.
+	configFiles, err := afero.ReadDir(options.Source, "/config/_default")
```
[the remark this answers is not in the document]
Comment (MashiroWang) [reply internal/static/identify.go]: of course, fixed.

File: public/categories/index.xml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
```
[the remark this answers is not in the document]
Comment (MashiroWang) [reply public/categories/index.xml]: sorry my fault.

File: internal/static/identify.go [hunk 3]
```diff
@@ -23,7 +22,15 @@ func (i *identify) PlanType() types.PlanType {
...
-	return utils.HasFile(fs, "index.html", "hugo.toml")
+	hasFile := utils.HasFile(fs, "index.html", "hugo.toml")
+	if !hasFile {
+		// Determining the configuration directory mode for Hugo.
+		// https://gohugo.io/getting-started/configuration/#configuration-directory
+		hugoConfigSource := afero.NewBasePathFs(fs, "config/_default")
+		hasHugoConfigFile := utils.HasFile(hugoConfigSource, "hugo.toml")
+		return hasHugoConfigFile
+	}
+	return hasFile
```
Comment (pan93412) [inline internal/static/identify.go:33]:
```suggestion
	return utils.HasFile(fs, "index.html", "hugo.toml", "config/_default/hugo.toml")
```

I did not verify if this works.

File: internal/static/identify.go [hunk 4]
```diff
@@ -32,6 +39,12 @@ func (i *identify) PlanMeta(options plan.NewPlannerOptions) types.PlanMeta {
 		return types.PlanMeta{"framework": "hugo"}
 	}
 
+	// Trying to locate the target file in the configuration directory.
+	hugoConfigSource := afero.NewBasePathFs(options.Source, "config/_default")
+	if utils.HasFile(hugoConfigSource, "hugo.toml") {
+		return types.PlanMeta{"framework": "hugo"}
+	}
```
Comment (pan93412) [inline internal/static/identify.go:46]:
```suggestion
	if utils.HasFile(hugoConfigSource, "config/_default/hugo.toml") {
		return types.PlanMeta{"framework": "hugo"}
	}
```

**Review by pan93412 — CHANGES_REQUESTED** [review]

File: internal/static/identify.go (hunk 3 above)
Comment (MashiroWang) [reply internal/static/identify.go]: perfect suggestion

File: internal/static/identify.go (hunk 4 above)
Comment (MashiroWang) [reply internal/static/identify.go]: perfect suggestion

**Comment by MashiroWang** [comment]
I think I can use glob pattern to match multiple Hugo config file extensions.

**Comment by MashiroWang** [comment]
> I think I can use glob pattern to match multiple Hugo config file extensions.

Afero glob unsupported list pattern like `{hugo,config}.{toml,yaml}`.
This solution is not feasible.

**Review by pan93412 — APPROVED** [review]

**Pull request merged**